helios-optics/src/main/kotlin/helios/optics/Optional.kt

```diff
@@ -0,0 +1,277 @@
+package helios.optics
+
+import arrow.*
+import arrow.core.*
+import arrow.optics.Fold
+import arrow.optics.Getter
+import arrow.optics.Iso
+import arrow.optics.PIso
+import arrow.optics.PLens
+import arrow.optics.PPrism
+import arrow.optics.PSetter
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Monoid
+import arrow.typeclasses.applicative
+
+
+
+inline fun <S, T, A, B> OptionalOf<S, T, A, B>.fix(): POptional<S, T, A, B> = this as POptional<S, T, A, B>
+
+typealias OptionalOf<S, T, A, B>  = Kind4<POptional.ForOptional, S, T, A, B>
+
+/**
+ * [Optional] is a type alias for [POptional] which fixes the type arguments
+ * and restricts the [POptional] to monomorphic updates.
+ */
+typealias Optional<S, A> = POptional<S, S, A, A>
+
+/**
+ * An [Optional] is an optic that allows to see into a structure and getting, setting or modifying an optional focus.
+ *
+ * A (polymorphic) [POptional] is useful when setting or modifying a value for a type with a optional polymorphic focus
+ * i.e. POptional<Ior<Int, Double>, Ior<String, Double>, Int, String>
+ *
+ * A [POptional] can be seen as a weaker [Lens] and [Prism] and combines their weakest functions:
+ * - `set: (S, B) -> T` meaning we can focus into an `S` and set a value `B` for a target `A` and obtain a modified source `T`
+ * - `getOrModify: (S) -> Either<T, A>` meaning it returns the focus of a [POptional] OR the original value
+ *
+ * @param S the source of a [POptional]
+ * @param T the modified source of a [POptional]
+ * @param A the focus of a [POptional]
+ * @param B the modified focus of a [POptional]
+ */
+interface POptional<S, T, A, B> : OptionalOf<S, T, A, B> {
```

Comment:
We are getting rid of this once Arrow 0.7.0 is released right?

Reply:
`Kind` is not enabled in Arrow-optics yet.